File: nodejs/7 kyu/99.js
// https://www.codewars.com/kata/559d7951ce5e0da654000073/train/javascript

// Passed

function alternateSqSum(arr) {
  const square_sums = arr.map((num, idx) => (idx % 2 !== 0 ? num ** 2 : num));
  const result = square_sums.reduce((prev, curr) => prev + curr, 0);
  return result;
}

const output = alternateSqSum([1, 2, 3, 4, 5]);
console.log(output);

File: python/6 kyu/99.py
# https://www.codewars.com/kata/59590976838112bfea0000fa/train/python

# Passed

def beggars(values, n):
    result = []
    for beggar_idx in range(n):
        beggar_collection = 0
        while beggar_idx < len(values):
            collection = values[beggar_idx]
            beggar_collection += collection
            beggar_idx += n
        result.append(beggar_collection)

    return result

output = beggars([1,2,3,4,5], 3)
print(output)